### PR TITLE
Initial RK3588 support in nixpkgs

### PR DESCRIPTION
**Pull request by thefossguy**
## Description of changes

Add initial support for the RK3588 and RK3588s SoCs and the SBCs that use them. I only own the Radxa Rock 5 Model B and Xunlong Orange Pi 5 so that is what I have added support for.

## Things done

1. Add initial support for the RK3588 SoC in arm-trusted-firmware by using the Collabora repository.
2. Introduce package `rkbin` that contains, among other proprietary blobs, the `ROCKCHIP_TPL` blob used to build U-Boot for the SBCs using this SoC.
4. Add U-Boot packages for two SBCs that I own: Radxa Rock 5 Mode B and Xunlong Orange Pi 5.

Tested:
1. Rock 5 Model B: U-Boot flashed to SD and also to the SPI can boot the NixOS ISO*.
2. Orange Pi 5: U-Boot flashed to SD can boot the NixOS ISO*.

*Support for RK3588 in the Linux kernel was added after 6.1 (6.4 IIRC) kernel, so I built a custom ISO with the 6.7.7 kernel and it works.

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/misc/arm-trusted-firmware/default.nix
Comment (samueldr) [inline pkgs/misc/arm-trusted-firmware/default.nix:36]: - Use `fetchFromGitLab`, which can also work with other instances than gitlab dot com.
 - Remove the `if` condition and pass `src` down where you define `armTrustedFirmwareRK3588`

The extra args given to `buildArmTrustedFirmware` allows overriding the default args.

**Comment by motiejus** [comment]
Good job!

Gentle reminder about https://nixos.wiki/wiki/NixOS_on_ARM/Orange_Pi_5 once this is in a good state.

File: pkgs/misc/uboot/default.nix
Comment (samueldr) [inline pkgs/misc/uboot/default.nix:417]: Is `u-boot-rockchip-spi.bin` not being produced by this build?

**Comment by thefossguy** [comment]
Uh oh, sorry for that... Thought I needed to target the staging branch, did that and saw a lot of commits unrelated to mine. Didn't intend to, sorry! (It's 03:44 past midnight here and I should've just gone to bed after creating this PR.)

File: pkgs/misc/uboot/default.nix
Comment (samueldr) [inline pkgs/misc/uboot/default.nix:35]: Can you extract the upgrade to 2024.01 in a different PR? This will need testing and validation on the existing builds too.

**Review by samueldr — DISMISSED** [review]

**Comment by thefossguy** [comment]
> Good job!
> 
> Gentle reminder about https://nixos.wiki/wiki/NixOS_on_ARM/Orange_Pi_5 once this is in a good state.

Too tired to understand what's happening or what I need to do but AFAIK, booting with both of these boards should be possible with steps laid out here: https://nixos.wiki/wiki/NixOS_on_ARM/UEFI#UEFI.2C_on_my_SBC.3F.3F.3F

File: pkgs/misc/uboot/default.nix
Comment (thefossguy) [reply pkgs/misc/uboot/default.nix:417]: The SPI binary is only built for the Rock 5 Model B, not even EVB. At least for v2024.01.

File: pkgs/misc/uboot/default.nix
Comment (thefossguy) [reply pkgs/misc/uboot/default.nix:35]: Understandable, I'll do that tomorrow :)

File: pkgs/misc/arm-trusted-firmware/default.nix
Comment (thefossguy) [reply pkgs/misc/arm-trusted-firmware/default.nix:36]: Understood, will tend to this tomorrow morning :)

Comment (thefossguy) [reply pkgs/misc/arm-trusted-firmware/default.nix:36]: On the topic of creating a separate PR, is it okay to introduce the `rkbin` package in this PR or should that be separate too?

File: pkgs/misc/uboot/default.nix
Comment (thefossguy) [reply pkgs/misc/uboot/default.nix:35]: #292757 is ready for review :)

File: pkgs/misc/arm-trusted-firmware/default.nix
Comment (thefossguy) [reply pkgs/misc/arm-trusted-firmware/default.nix]: Does https://github.com/NixOS/nixpkgs/pull/292667/commits/c86043fd665093bb9a11fa52aef410536272289f look like what you asked?

File: pkgs/by-name/rk/rkbin/package.nix
Comment (thefossguy) [inline pkgs/by-name/rk/rkbin/package.nix:8]: @samueldr do you have any comments on this? There really isn't any version number in upstream other than just the commit history. I believe that this is the best approach from what I know but I'd love to hear your thoughts!

**Comment by adminy** [comment]
Actually @thefossguy I noticed there are some new versions used and tested:

https://github.com/nabam/nixos-rockchip/blob/main/pkgs/uboot-rockchip.nix#L83-L84

**Comment by thefossguy** [comment]
@adminy I don't quite understand what new versions you are referring to... This PR is only for introducing support for _one of the [many] Rockchip SoCs_: the RK3588 series of SoCs in particular. This was done because I own two SBCs that use an SoC from said lineup.

Maybe you are referring to supporting boards supported by the repository you listed? In which case, I'd be a bit hesitant in adding support for those **as I do not own any of those boards**. That said, I'd be happy to maintain it, _given that someone provides me with initial feedback on what's working and what isn't_.

**Comment by adminy** [comment]
Yep, that's what I was trying to say. I have nixos running with the RK3568 SOC (cm4 soquartz). My Config to get it working looks like so:
```nix
let rkbin = pkgs.fetchFromGitHub {
	owner = "rockchip-linux";
        repo = "rkbin";
        rev = "a2a0b89b6c8c612dca5ed9ed8a68db8a07f68bc0";
        sha256 = "U/jeUsV7bhqMw3BljmO6SI07NCDAd/+sEp3dZnyXeeA=";
  };
 buildPatchedUBoot = { defconfig, BL31, ROCKCHIP_TPL ? "" }: pkgs.buildUBoot {
      inherit defconfig BL31 ROCKCHIP_TPL;
      enableParallelBuilding = true;
      src = pkgs.fetchFromGitHub {
        owner = "u-boot";
        repo = "u-boot";
        rev = "v2024.04-rc3";
        sha256 = "gJd5mVACstzLmUMv9sIR8YMR6neBcARBTUwv8tSAUJ0=";
      };
      version = "v2024.04-rc3-d49fa3d";
      filesToInstall = [ "u-boot-rockchip.bin" ];
      patches = [
        (pkgs.fetchpatch {
          name = "quartz64.patch";
          url = "https://github.com/Kwiboo/u-boot-rockchip/compare/d49fa3defa50c6d3f04acbb52fd486c13c14ab6a...ced42f309e8408d0e471e9b98b467942226667c7.diff";
          sha256 = "/ejaiMmMxfnvvS8Hl4QGZ4fkVGIzdYFWfL1zbqQFnds=";
        })
      ];

      buildInputs = with pkgs.buildPackages; [
        ncurses # tools/kwboot
        libuuid # tools/mkeficapsule
        gnutls # tools/mkeficapsule
        openssl # tools/imagetool
      ];

      nativeBuildInputs = with pkgs.buildPackages; [
        ncurses # tools/kwboot
        bc
        bison
        dtc
        flex
        openssl
        (python3.withPackages (p: [
          p.libfdt
          p.setuptools # for pkg_resources
          p.pyelftools
        ]))
        swig
        which # for scripts/dtc-version.sh
      ];

      makeFlags = [ "CROSS_COMPILE=${pkgs.stdenv.cc.targetPrefix}" ];
      extraMeta = {
        platforms = [ "aarch64-linux" ];
        license = lib.licenses.unfreeRedistributableFirmware;
      };
    };
    uBoot = buildPatchedUBoot {
      defconfig = "soquartz-cm4-rk3566_defconfig";
      BL31 = (rkbin + "/bin/rk35/rk3568_bl31_v1.44.elf");
      ROCKCHIP_TPL = (rkbin + "/bin/rk35/rk3566_ddr_1056MHz_v1.21.bin");
    };
```

**Comment by thefossguy** [comment]
@adminy can you try building `rkbin` with the following patch and let me know if it works out for you or not?

```diff
diff --git a/pkgs/by-name/rk/rkbin/package.nix b/pkgs/by-name/rk/rkbin/package.nix
index fb45fa9f53ef..f0105e2423f8 100644
--- a/pkgs/by-name/rk/rkbin/package.nix
+++ b/pkgs/by-name/rk/rkbin/package.nix
@@ -24,6 +24,9 @@ stdenv.mkDerivation rec {

   ROCKCHIP_TPL_RK3588 = "bin/rk35/rk3588_ddr_lp4_2112MHz_lp5_2736MHz_v1.12.bin";

+  BL31_RK3568 = "bin/rk35/rk3568_bl31_v1.43.elf";
+  ROCKCHIP_TPL_RK3568 = "bin/rk35/rk3566_ddr_1056MHz_v1.18.bin";
+
   meta = with lib; {
     description = "Rockchip proprietary bootloader blobs";
     homepage = "https://github.com/rockchip-linux/rkbin";
```

> Quick question: Is Orange Pi 5 Plus something that could be easily added with this PR or Pi 5 and Pi 5 Plus are very different?

I can add _all_ RK3588 boards, it's quite easy! But I didn't because I only have the two boards I mentioned. I can only add support for a board after I have tested the U-Boot package on it or someone else does it for me.

Since you are interested, I can add support for the Orange Pi 5 Plus, **given that you test normal functionality** (testing NVMe boot would be quite nice!). Please apply following patch to my tree (mind the dependency on #277997), build, test and let me know if it works out for you or not! Thank you :)

```diff
diff --git a/pkgs/misc/uboot/default.nix b/pkgs/misc/uboot/default.nix
index 580277f327c9..3e37d7b228f0 100644
--- a/pkgs/misc/uboot/default.nix
+++ b/pkgs/misc/uboot/default.nix
@@ -417,6 +417,14 @@ in {
     filesToInstall = [ "u-boot.itb" "idbloader.img" "u-boot-rockchip.bin" ];
   };

+  ubootOrangePi5Plus = buildUBoot {
+    defconfig = "orangepi-5-plus-rk3588_defconfig";
+    extraMeta.platforms = ["aarch64-linux"];
+    BL31 = "${armTrustedFirmwareRK3588}/bl31.elf";
+    ROCKCHIP_TPL = "${rkbin}/${rkbin.ROCKCHIP_TPL_RK3588}";
+    filesToInstall = [ "u-boot.itb" "idbloader.img" "u-boot-rockchip.bin" ];
+  };
+
   ubootPcduino3Nano = buildUBoot {
     defconfig = "Linksprite_pcDuino3_Nano_defconfig";
     extraMeta.platforms = ["armv7l-linux"];
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index 74ea8f383f8d..21e84b95edb3 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -28554,6 +28554,7 @@ with pkgs;
     ubootOlimexA64Teres1
     ubootOrangePi3
     ubootOrangePi5
+    ubootOrangePi5Plus
     ubootOrangePiPc
     ubootOrangePiZeroPlus2H5
     ubootOrangePiZero
```

**Comment by adminy** [comment]
```diff
diff --git a/pkgs/by-name/rk/rkbin/package.nix b/pkgs/by-name/rk/rkbin/package.nix
index fb45fa9f53ef..f0105e2423f8 100644
--- a/pkgs/by-name/rk/rkbin/package.nix
+++ b/pkgs/by-name/rk/rkbin/package.nix
@@ -24,6 +24,9 @@ stdenv.mkDerivation rec {

   ROCKCHIP_TPL_RK3588 = "bin/rk35/rk3588_ddr_lp4_2112MHz_lp5_2736MHz_v1.12.bin";

+  BL31_RK3568 = "bin/rk35/rk3568_bl31_v1.43.elf";
+  ROCKCHIP_TPL_RK3568 = "bin/rk35/rk3566_ddr_1056MHz_v1.18.bin";
+
   meta = with lib; {
     description = "Rockchip proprietary bootloader blobs";
     homepage = "https://github.com/rockchip-linux/rkbin";
```
This does not work, also I had to change the revision:
```nix
let rkbin = pkgs.fetchFromGitHub {
  owner = "rockchip-linux";
  repo = "rkbin";
  rev = "70a4b3fc75053d10a2718a75591a2fe998a207de"; #"b4558da0860ca48bf1a571dd33ccba580b9abe23";
  hash = "sha256-ndnx/gzWkxuZ9g+l90ev9Dsai34H4SR1VewYzQpXzz8=";
  # hash = "sha256-KUZQaQ+IZ0OynawlYGW99QGAOmOrGt2CZidI3NTxFw8=";
 };
```
Because they were not available in the commented out revision. Looks like the newer ones are working though.
Its interesting because I got these to work before:
```
ROCKCHIP_TPL = "/bin/rk35/rk3566_ddr_1056MHz_v1.16.bin";
BL31 = "/bin/rk35/rk3568_bl31_v1.42.elf";
```


I'll try the 5 Plus next. That one's going to take a while to set up with the nvme and everything. Hopefully will get it to work by the end of today.

**Comment by motiejus** [comment]
I tried building `ubootOrangePi5Plus` and got this:

```
error: build of '/nix/store/lva3xa9g6qfl8hvi0bawg2j9965plgwf-uboot-orangepi-5-plus-rk3588_defconfig-2024.01.drv' on 'ssh-ng://remote-builder@100.89.176.5' failed: builder for '/nix/store/lva3xa9g6qfl8hvi0bawg2j9965plgwf-uboot-orangepi-5-plus-rk3588_defconfig-2024.01.drv' failed with exit code 2;
       last 10 log lines:
       >   AR      lib/built-in.o
       >   LD      u-boot
       >   OBJCOPY u-boot.srec
       >   OBJCOPY u-boot-nodtb.bin
       >   SYM     u-boot.sym
       >   RELOC   u-boot-nodtb.bin
       > make[2]: *** No rule to make target 'arch/arm/dts/rk3588-edgeble-neu6a-io.dtb', needed by 'dtbs'.  Stop.
       > make[1]: *** [dts/Makefile:44: arch-dtbs] Error 2
       > make: *** [Makefile:1165: dts/dt.dtb] Error 2
       > make: *** Waiting for unfinished jobs....
       For full logs, run 'nix log /nix/store/lva3xa9g6qfl8hvi0bawg2j9965plgwf-uboot-orangepi-5-plus-rk3588_defconfig-2024.01.drv'.
error: builder for '/nix/store/lva3xa9g6qfl8hvi0bawg2j9965plgwf-uboot-orangepi-5-plus-rk3588_defconfig-2024.01.drv' failed with exit code 1
```

I tried updating uboot and rkbin as specify by @adminy , but couldn't get anywhere (looks like may still be some obsolete hashes or versions).

I brought up my Orange Pi 5+ to the state where I can test this. Looking forward to an updated PR.

**Comment by motiejus** [comment]
For the record, this combination of rkbin, uboot and tlp boots the kernel from NVME on orangepi5plus: https://git.jakstys.lt/motiejus/config/src/commit/8d0031e6916f8e4fb812eb61ee13f3f205d21cce/hacks/orangepi5plus/uboot/default.nix

I am working to get NixOS to mount the root filesystem, but I believe it's a problem way higher up the stack and thus out of scope of your PR.

File: pkgs/by-name/rk/rkbin/package.nix
Comment (flokli) [inline pkgs/by-name/rk/rkbin/package.nix:10]: Can you move these two into an outer let binding, so they're not bleeding as env vars into the build itself? Or do these need to be set in the build? If so, a comment on top would be good.

File: pkgs/by-name/rk/rkbin/package.nix
Comment (flokli) [inline pkgs/by-name/rk/rkbin/package.nix:25]: Do you need to compile a different rkbin, depending on the board type? Or why does this encode a specific board type here?

File: pkgs/misc/arm-trusted-firmware/default.nix
Comment (flokli) [inline pkgs/misc/arm-trusted-firmware/default.nix:160]: Is there any upstream CL that we could link to in the comment?

File: pkgs/misc/uboot/default.nix
Comment (flokli) [reply pkgs/misc/uboot/default.nix:35]: Let's mark this PR as draft then until that PR landed, so it doesn't get merged by accident.

**Review by flokli — COMMENTED** [review]
Awesome! I'm also trying to bring up some rk3588 boards (turing rk1), so thanks for this.

**Comment by thefossguy** [comment]
@motiejus this PR depends on #277997 :)

File: pkgs/by-name/rk/rkbin/package.nix
Comment (thefossguy) [reply pkgs/by-name/rk/rkbin/package.nix:10]: Sure, I can move them in a let binding outside since those don't actually need to be in the build itself.

File: pkgs/misc/arm-trusted-firmware/default.nix
Comment (flokli) [reply pkgs/misc/arm-trusted-firmware/default.nix:160]: Arm trusted firmware uses Gerrit, that's their unit of code review, compared to PRs on GitHub.

File: pkgs/by-name/rk/rkbin/package.nix
Comment (flokli) [reply pkgs/by-name/rk/rkbin/package.nix:25]: But does this make this `rkbin` one that's only applicable for rk3588?

File: pkgs/misc/arm-trusted-firmware/default.nix
Comment (thefossguy) [reply pkgs/misc/arm-trusted-firmware/default.nix:160]: I'm aware of that, though I still don't know what the term "CL" is

Comment (flokli) [reply pkgs/misc/arm-trusted-firmware/default.nix:160]: It means "changelist". See https://commondatastorage.googleapis.com/chrome-infra-docs/flat/depot_tools/docs/html/git-cl.html for a reference to it.

**Comment by thefossguy** [comment]
@adminy this PR needs to be rebased and some things need to be cleaned up. I'll do that once #292757 is merged :)

**Comment by flokli** [comment]
> @adminy this PR needs to be rebased and some things need to be cleaned up. I'll do that once #292757 is merged :)

The PR got merged, can you rebase this PR (and retarget staging)?

**Comment by flokli** [comment]
@thefossguy You can't target master now, as doesn't have the uboot bump from https://github.com/NixOS/nixpkgs/pull/292757.

You'd need to wait for this to [bubble through an entire staging cycle and all the channels](https://nixpk.gs/pr-tracker.html?pr=292757) buildUBootwhich would take quite some time.

That's why I'm proposing to have this target staging too, so ideally both end up in the same staging cycle.

Another alternative would be to remove all introduced package attributes calling `buildUBoot` (effectively reducing this PR to `rkbin` and `armTrustedFirmwareRK3588`), merging them to staging, and introducing the different u-boot builds once the staging cycle completed (in a separate PR).

Note I manually cherry-picked your changes from here into my own config-repo, and got the Turing RK1 to boot (using a slightly more recent, not-yet-released u-boot - but with the same patch that landed) - so all the `rkbin` and arm TF bits seem to work.

I think this PR should be quite uncontroversial to merge to staging, once it's rebased.

**Comment by K900** [comment]
I don't know if this is worth doing tbh, but maybe we should have multiple uboot variants for boards with mixed mode (SATA/PCIe) M.2 slots? I think the Rock5B only has PCIe wired up, but the OPi5 has both and relies on software (yes, yuck) to mux them. Ideally uboot would probe both in order, but that's too much for now.

**Comment by flokli** [comment]
> I don't know if this is worth doing tbh, but maybe we should have multiple uboot variants for boards with mixed mode (SATA/PCIe) M.2 slots? I think the Rock5B only has PCIe wired up, but the OPi5 has both and relies on software (yes, yuck) to mux them. Ideally uboot would probe both in order, but that's too much for now.

The important part IMHO is to have the basic tooling up, and some usage examples. If you decide to call `buildUBoot` slightly differently, that's only a small variation that could easily live outside nixpkgs if not too common.

File: pkgs/by-name/rk/rkbin/package.nix
Comment (thefossguy) [reply pkgs/by-name/rk/rkbin/package.nix:13]: Thanks for pointing it out, I'll make that change. Just to see if I understand correctly, the version string should be `unstable-2024.02.22`, right?

File: pkgs/by-name/rk/rkbin/package.nix
Comment (thefossguy) [reply pkgs/by-name/rk/rkbin/package.nix:18]: ack

File: pkgs/by-name/rk/rkbin/package.nix
Comment (thefossguy) [reply pkgs/by-name/rk/rkbin/package.nix:13]: I see, will re-push. Thank you :)

**Comment by thefossguy** [comment]
I'm going to be busy for a few weeks from today and want to get this merged before that. So, @flokli and @samueldr, are there any changes you would like me to make to any commits in this PR? (Other than the `unstable-` prefix for `rkbin`.)

Tonight's force-push will be the final one until I am free from the internship and completion of my degree :)

**Comment by flokli** [comment]
> I'm going to be busy for a few weeks from today and want to get this merged before that. So, @flokli and @samueldr, are there any changes you would like me to make to any commits in this PR? (Other than the `unstable-` prefix for `rkbin`.)

That was the only thing I was waiting for to push the button ;-)

**Comment by flokli** [comment]
Successfully built the two new uboot targets after freshly merging this into staging locally. Thanks!

File: pkgs/by-name/rk/rkbin/package.nix
Comment (wegank) [inline pkgs/by-name/rk/rkbin/package.nix:31]: @thefossguy Could you add yourself to the list of maintainers in another PR targeting staging? Many thanks.

Comment (wegank) [reply pkgs/by-name/rk/rkbin/package.nix:31]: Removed in #297511 temporarily.

Comment (flokli) [reply pkgs/by-name/rk/rkbin/package.nix:31]: Oof, I am sorry for not seeing this until now.

Comment (thefossguy) [reply pkgs/by-name/rk/rkbin/package.nix:31]: @wegank stupid question: I will create a new PR that adds me as the maintainer, but for that, should I still target the `staging` branch (which this PR targeted)?

Comment (K900) [reply pkgs/by-name/rk/rkbin/package.nix:31]: You can send it directly to staging, or send it to master and it'll get auto-merged into staging in a few hours.

Comment (thefossguy) [reply pkgs/by-name/rk/rkbin/package.nix:31]: Done in #297643.

**Comment by K900** [comment]
How did this end up in staging, anyway? It looks like it can go straight to master.

**Comment by flokli** [comment]
> How did this end up in staging, anyway? It looks like it can go straight to master.

https://github.com/NixOS/nixpkgs/pull/277997 got retargeted to staging, rebuilding all kernels seemed to be too much for master ?!

**Comment by K900** [comment]
But that one is in master already. I guess this just missed the staging-next merge? Anyway we can also merge this into master.

**Comment by flokli** [comment]
Yeah, this can be cherry-picked to there now too, but I didn't want to say in https://github.com/NixOS/nixpkgs/pull/292667#issuecomment-2003706651 to wait some more days, but rather land this.

**Pull request merged**
